src/stores/game.js

```diff
@@ -285,6 +285,12 @@ export const useGameStore = defineStore('game', {
     },
     opponentLeft() {
       this.players = this.players.filter((player) => player.pNum === this.myPNum);
+      
+      if(this.myPNum == 0) {
+        this.p1Ready = false;
+      } else {
+        this.p0Ready = false;
+      }
```

Comment:
We should investigate if the failure is in the frontend or the backend. This will force a given player's game to reflect that they are not ready when they first join, but do we know if the problem is persisted in the database or only in the frontend? We can use a combination of the vue devtools, backend blueprint routes, and a debugger to explore this.

Something to test: 
If you create a game and go through the process that replicates the bug then go to `http://localhost:1337/game` and look at the game you just made, what are the values of `p0Ready` and `p1Ready` for that game? This will show the mock database representation for all the games, so we can compare that value to the one we see in the vue devtools to see whether the frontend and backend agree (and are _both_ wrong) or the issue is only in the frontend

Reply:
I wrote new tests for the ready/notReady/leave/joinBack sequences and it seems like the bug is in the client.  The server is always spot on. It keeps track of what's going on in the two clients.  
The 'leftGame' socket connection defined in \src\plugins\sails.js calls 'handleLeftGame' plugins\gameListEvents.js, wich calls opponentLeft() in stores.game.js.  This method only took the opponent out of the state.players array.  So the pOReady/p1Ready was not updated. That's where I add a few lines to do so.

Other than that, if the opponent readies and leave, and then the (other) player leaves and come back, both indicators display the player's name, not the oponent's name.  When tested in production (the online version) the same sequence would end up with the opponent' indicator staying empty. No user name and no card.  It seems related... but should I open another issue?

Reply:
Okay in that case let's remove this cleanup from here and instead clear out _both_ `p0Ready` and `p1Ready`when the player leaves the lobby, inside the `requestLeaveLobby` action in this same file